### PR TITLE
Adding the go tutorial

### PR DESCRIPTION
**Pull request by chrislovecnm**
This tutorial covers using the ruleset rules_go and gazelle.

### CODE REVIEW

File: go-tutorial.md [hunk 1]
```diff
@@ -0,0 +1,1233 @@
...
+This tutorial is going to cover how bazel support the programming language Go.
+
+[Bazel](https://bazel.build) is an open-source build and test application that supports the software development lifecycle. Bazel strives
+to allow a developer to have hermetic and deterministic builds, testing, packaging and deployment.  This build
+tool supports multiple languages and builds binaries for multiple platforms.
+
+Of the multiple languages that bazel supports includes [Go](https://go.dev). Go is an open-source programming languages
+that was created by Google.
+
+One of the concepts within bazel is a rule.  A bazel rule defines a series of actions and outputs. Toolchains is another
```
Comment (cgrindel) [inline go-tutorial.md:32]: Outside of a code snippet, should `bazel` always be capitalized (e.g., `Bazel`)?

File: go-tutorial.md [hunk 2]
```diff
@@ -0,0 +1,1233 @@
...
+
+- go: https://go.dev/doc/install
+- gcc: use your systems package manager
+- bazelisk: https://github.com/bazelbuild/bazelisk#installation
+
+Technically we do not need the go binary installed, but we are going to use
+`cobra-cli` to generate some project code.  We did not want to add the 
+extract work in bazel to run the binary using bazel. A developer, using go,
+does not need to download the go binary. In order to keep a build deterministic
+bazel and rules_go download go. rules_go require that gcc is installed.
```
Comment (cgrindel) [inline go-tutorial.md:88]: I get not wanting to set up `cobra-cli` as a dep of the Bazel workspace. What do you think about using the Go SDK downloaded by `rules_go` to keep things reasonably hermetic?

The steps would be something like:
1. Create the `WORKSPACE` as you have described, except do not include the loading of the `deps.bzl` file.
2. Init the module: `bazel run @go_sdk//:bin/go -- mod init github.com/chrislovecnm/bazel-go-example-code`
3. Install `cobra-cli`: `bazel run @go_sdk//:bin/go -- install github.com/spf13/cobra-cli@latest`
4. Perform your `cobra-cli` steps: `$HOME/go/bin/cobra-cli init -a '${NAME}' --license apache`

All of that being said, I would be amazed that someone walking through the tutorial does not already have `go` installed. What I think is interesting about using the downloaded SDK is that it shows off the hermetic toolchain.

Just throwing it out there.

**Comment by cgrindel** [comment]
Thanks for writing this up. This is great!

File: go-tutorial.md (hunk 2 above)
Comment (chrislovecnm) [reply go-tutorial.md]: I think it is pretty cool, yes it shows how to use the go rules more.  Let me see how it works out.  I would like to make this an issue and update the tutorial after we get this merged.  Thoughts?

**Comment by chrislovecnm** [comment]
https://github.com/bazel-contrib/Bazel-learning-paths/pull/6 this PR supports this documentation.

File: go-tutorial.md (hunk 2 above)
Comment (cgrindel) [reply go-tutorial.md]: That is fine with me.

File: go-tutorial.md (hunk 1 above)
Comment (chrislovecnm) [reply go-tutorial.md]: Updated both the words "Bazel" and "Gazelle"

**Review by cgrindel — APPROVED** [review]

**Comment by chrislovecnm** [comment]
@cgrindel I need to update with more edits.

**Pull request merged**